test/test_cuda.py

```diff
@@ -1099,6 +1100,14 @@ def test_cat_bad_input_sizes(self):
         z = torch.randn(2, 2, 1).cuda()
         self.assertRaises(RuntimeError, lambda: torch.cat([x, y, z], dim=1))
 
+    @unittest.skipIf(torch.cuda.device_count() >= 10, "Loading a cuda:9 tensor")
+    @unittest.skipIf(not PY3, "Tensor was serialized with Python 3")
+    def test_load_nonexistent_device(self):
+        msg = r'Attempting to deserialize object on CUDA device 9'
+        with self.assertRaisesRegex(RuntimeError, msg):
+            fname = os.path.join(os.path.dirname(__file__), 'data/pi-cuda9.pt')
```

Comment:
Hmm is there any way to fake saving this without checking in those binary files in the repo? `git` is pretty bad at handling those.

Reply:
Possibly, I could write an output file on CPU and then replace all instances of 'cpu' with 'cuda:0' and/or 'cuda:9'. Let me try that and see how it goes